macros/src/lib.rs

```diff
@@ -65,7 +66,19 @@ pub fn statemachine(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
                 return e.to_compile_error().into();
             }
 
-            codegen::generate_code(&sm).into()
+            let tokens = codegen::generate_code(&sm).into();
+            // Write expanded state machine to a file for reference.
+            let name = if let Some(name) = &sm.name {
+                name.to_string()
+            } else {
+                "default".to_string()
+            };
+            let mut expansion_file =
+                std::fs::File::create(format!("target/smlang-expansion-{name}.rs")).unwrap();
+            expansion_file
+                .write_all(format!("{tokens}").as_bytes())
+                .unwrap();
+            tokens
```

Comment:
I found myself continuously trying to find the expanded code, so I ended up writing this in similar to RTIC. I need to check if this is the write way to get it into the target build directory.

Reply:
Removed this for now. We can add it on a follow-up PR. The target path derivation is more complex than this.